ir/memory.cpp

```diff
@@ -248,16 +248,37 @@ pair<expr, vector<expr>> Memory::mkInput(const char *name) {
   return { Pointer(*this, offset, local_bid, bid).release(), { var } };
 }
 
-expr Memory::alloc(const expr &size, unsigned align, bool heap) {
-  Pointer p(*this, ++last_bid, true);
+expr Memory::alloc(const expr &size, unsigned align, BlockKind blockKind,
+                   optional<unsigned> bidopt, unsigned *bid_out) {
+  unsigned bid = bidopt ? *bidopt : last_bid;
+  if (last_bid <= bid) {
```

Comment:
you do a bump in tgt, and in src bidopt should be null. I would ensure this invariant with:
```
if (!bidopt)
  ++last_bid;
assert(bid < last_bid);
```

Reply:
I see, updated.